tests/src/Examples.hs

```diff
@@ -0,0 +1,109 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Examples (solverTests, processTests) where
+
+import qualified Data.ByteString.Lazy.Char8 as LBS
+import Data.Default (def)
+import SMTLIB.Backends (Backend, command, command_, initSolver)
+import qualified SMTLIB.Backends.Process as Process
+import System.Exit (ExitCode (ExitSuccess))
+import System.IO (BufferMode (LineBuffering), hSetBuffering)
+import System.Process.Typed (getStdin)
+import Test.Tasty
+import Test.Tasty.HUnit
+
+-- | The examples for the `Backends` module.
+solverTests :: TestTree
+solverTests =
+  testGroup
+    "API use examples"
+    [ testCase "basic use" solverBasicUse
+    ]
+
+-- | Basic use of the `Solver` datatype.
+solverBasicUse :: IO ()
+solverBasicUse =
+  -- we need a solver backend to evaluate our commands
+  -- examples on how to create such backends can be found in the corresponding
+  -- sections, e.g. `processBasicUse`
+  withBackend $ \backend -> do
+    -- then, we create a solver out of the backend
+    -- we enable queuing (it's faster !)
+    solver <- initSolver backend True
+    -- we send a basic command to the solver and ignore the response
+    -- we can write the command as a simple string because we have enabled the
+    -- OverloadedStrings pragma
+    _ <- command solver "(get-info :name)"
+    return ()
+  where
+    withBackend :: (Backend -> IO a) -> IO a
+    withBackend = Process.with def . (. Process.toBackend)
```

Comment:
`solverBasicUse` does the same as `processBasicUse` but explains less about defaults. I think `solverBasicUse` could be removed and the user would learn as much with less reading.

Reply:
I did this to factor the explanation. Right now `Process` is the only backend, but if we add more backends one day then the information shouldn't be located in the examples of only one of the backends. But I guess this is premature optimization.

I feel that we can always write the readme so that the reader is first redirected to the `processBasicUse` example, and then if need be also mention the `solverBasicUse` one.

Reply:
I ended up removing this following the merge of #21 as I was weary `smtlib-backends`' test-suite depending on `smtlib-backends-process` (for the backend) would create another dependency cycle.

Another option could be adding an example on how to create a dummy backend, which would also motivate this separate example. But I'm not sure it would add much value as it would only show that we really need a function of type `Builder -> LBS.ByteString` that's able to return `success` when needed.